doc/source/development/roadmap-indexing-views.rst

```diff
@@ -0,0 +1,310 @@
+
+.. code-block:: python
+
+   # Case 1: user wants mutations of df2 to be reflected in df
+   >>> df = pd.DataFrame({"A": [1, 2], "B": [3, 4]})
+   >>> df2 = df[["A"]].as_mutable_view() # name TBD
+   >>> df2.iloc[:, 0] = 10 >>> df.iloc[0, 0] # df was mutated 10
+
+For the user who wishes to not mutate the parent, we require that the user
+explicitly break the reference from ``df2`` to ``df`` by implementing “Error on Write”.
+
+.. code-block:: python
+
+   # Case 2: The user does not want mutating df2 to mutate df, via EoW
+   >>> df = pd.DataFrame({"A": [1, 2], "B": [3, 4]})
+   >>> df2 = df[["A"]]
+   >>> df2.iloc[0, 0] = 10
+   ValueError("error on write to subset of other dataframe")
+   >>> df2 = df2.copy_if_needed() # API is TBD. Could be a keyword argument to copy.
+   >>> df2.iloc[:, 0] = 10 >>> df.iloc[0, 0] # df was not mutated 1
+
+Copy-on-Write vs. Error-on-Write
+--------------------------------
+
+Consider the following example:
+
+.. code-block:: python
+
+   >>> df2 = df[['A']]
+   >>> df2.iloc[0, 0] = 10 # df2 can be a view of df, what happens by default?
+   >>> df3 = df[df['A'] == 1]
+   >>> df3.iloc[0, 0] = 10 # df3 is already a copy of df, what happens by default?
+
+We have a few options for the default:
+
+1. Well-Defined copy/view rules: ensure we have more consistent rules (e.g.
+   selecting columns is always a view), and then views result in mutating the
+   parent, copies not. This comes down to fixing some bugs and clearly
+   documenting and testing which operations are views, and which are copies.
+2. Copy-on-Write: The setitem would check if it’s a view on another dataframe.
+   If it is, then we would copy our data before mutating.
+3. Error-on-Write: The setitem would check if it’s a subset of another dataframe
+   (both view of copy). Only rather than copying in case of a view we would
+   raise an exception telling the user to either copy the data with
+   ``.copy_if_needed()`` (name TBD) or mark the frame as “a mutable view” with
+   ``.as_mutable_view()`` (name TBD).
+
+We propose "Error on Write" by default. This forces a decision on the user, and
+is the most explicit in terms of code.
+
+Additionally, consider the "classic" case of chained indexing, which was the
+original motivation for the ``SettingWithCopy`` warning
+
+.. code-block:: python
+
+   >>> df[df['B'] > 4]['B'] = 10
+
+That is roughly equivalent to
+
+.. code-block:: python
+
+   >>> df2 = df['B'] > 4] # Copy under NumPy’s rules
+   >>> df2['B'] = 10 # Update (the copy) df2, df not changed
+   >>> del df2 # All references to df2 are lost, goes out of scope
+
+And so ``df`` is not modified. If we adopted Copy On Write to completely replace the
+current ``SettingWithCopy`` warning, we would restore the old behavior of silently
+“failing” to update ``df2``. Under Error on Write, we’d track that the ``df2`` created
+by the first getitem references ``df`` and raise an exception when it was being
+mutated.
```

Comment:
It may be worth including what error-on-write would be "roughly equivalent to" (including comments on each line) to make it clear what would happen if you did break things up like you did in lines 137-139